src/main/java/com/ekart/service/ProductService.java

```diff
@@ -44,7 +57,55 @@ public Product getProductById(Long id) throws RecordNotFoundException {
         }
     }
 
-    public Product saveProduct(Product product) {
+    public Product saveProduct(Map<String, Object> request) {
+        Category newCategory;
+        Brand newBrand;
+        Set<Size> newSize = new HashSet<>();
+        List<String> desc = new ArrayList<>();
+
+        String categoryName = (String) request.get(Constants.CATEGORY);
+        Optional<Category> category = categoryRepository.findCategoryByName(categoryName);
+        if(!category.isPresent()) {
+            newCategory = new Category();
+            newCategory.setName(categoryName);
+        } else {
+            newCategory = category.get();
+        }
+        String brandName = (String) request.get(Constants.BRAND);
+        Optional<Brand> brand = brandRepository.findBrandByName(brandName);
+        if(!brand.isPresent()) {
+            newBrand = new Brand();
+            newBrand.setName(brandName);
+        } else {
+            newBrand = brand.get();
+        }
+        String sizeName = (String) request.get(Constants.SIZE);
+        if(sizeName != null && sizeName.length() > 0) {
+            Optional<Size> size = sizeRepository.findSizeBySize(sizeName);
+            if(!size.isPresent()) {
+                Size sizeAdd = new Size();
+                sizeAdd.setSize(sizeName);
+                newSize.add(sizeAdd);
+            } else {
+                newSize.add(size.get());
+            }
+        }
+        String description = (String) request.get(Constants.DESCRIPTION);
+        if(description != null) {
+            desc = Arrays.stream(description.split("\n")).collect(Collectors.toList());
+        }
+        String color = (String) request.get(Constants.COLOR);
+        Color newColor = Color.valueOf(color);
+        byte[] thumbnailImage = ((String) request.get(Constants.THUMBNAIL)).getBytes();
+        List<byte[]> images = ((List<String>) request.get(Constants.IMAGES))
+                                .stream().map(data -> data.getBytes()).collect(Collectors.toList());
+        String name = (String) request.get(Constants.NAME);
+        BigDecimal price = new BigDecimal((Integer) request.get(Constants.PRICE));
+        Integer quantity = (Integer) request.get(Constants.QUANTITY);
+        Double discount = ((Integer) request.get(Constants.DISCOUNT)).doubleValue();
```

Comment:
The problem with using a map or object is we need to typecast for each property. Will see some other ways if exists to fix this problem..

Reply:
Alternative would be to create separate dto, similar to the request. But we cant use the modalmapper to map the dto to entity